image/helpers.go

```diff
@@ -404,3 +405,25 @@ func (tsto *ToolingStore) Find(at *asserts.AssertionType, headers map[string]str
 	}
 	return tsto.sto.Assertion(at, pk, tsto.user)
 }
+
+// var so that it can be mocked for tests
+var writeResolvedContent = writeResolvedContentImpl
+
+// XXX: move to gadget?
+func writeResolvedContentImpl(targetdir, gadgetUnpackDir, kernelUnpackDir string, model *asserts.Model) error {
+	lv, err := gadget.LaidOutSystemVolumeFromGadget(gadgetUnpackDir, kernelUnpackDir, model)
+	if err != nil {
+		return err
+	}
+	for _, ps := range lv.LaidOutStructure {
```

Comment:
for UC20 this code will write files that are not needed, because we really need to take care only of system-seed, no? OTOH maybe it's ok because is ready if we do some of install work already at ubuntu-image time later.

ATM the layouts are already different:

UC16/18
```
<prepare-dir>/image
```

UC20
```
<prepare-dir>/system-seed
```

Reply:
Yeah, this will produce too much data on uc20 currently (it. should stop after it found the system-seed).

Reply:
It maybe okay for now that prepare-image just outputs all the resolved content and ubuntu-image picks what it needs. Longer term ubuntu-image will be in go and it will call better and richer APIs than a (very limited) CLI itnerface.